src/util/StreamUtils.h

```diff
@@ -0,0 +1,110 @@
+//   Copyright 2023, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#ifndef QLEVER_STREAMUTILS_H
+#define QLEVER_STREAMUTILS_H
+
+#include <ctre/ctre.h>
+
+#include <boost/lexical_cast.hpp>
+
+#include "util/Exception.h"
+
+namespace ad_utility {
+
+// Wrapper type for std::chrono::duration<> to avoid having to declare
+// this in the std::chrono namespace.
+template <typename DurationType>
+class ParseableDuration {
+  DurationType duration_{};
+
+  template <typename CharT>
+  friend inline std::basic_istream<CharT>& operator>>(
+      std::basic_istream<CharT>& is, ParseableDuration<DurationType>& result) {
+    using namespace std::chrono;
+    std::basic_string<CharT> arg;
+
+    is >> arg;
```

Comment:
Shouldn't this just be a function that takes `std::string_view` and returns `DurationType`?
Then you don't need the ParseableDuration wrapper.

(Those are two things,  i don't see why you need `istream`, because you read to a string directly. )

The output function can then maybe be an `ostream<<` , but maybe also just a `toString` function.

Reply:
Maybe only two fr4ee (template) functions `durationToString` and `durationFromString` are much easier to handle.